nibabel/gifti/parse_gifti_fast.py

```diff
@@ -168,13 +168,14 @@ def StartElementHandler(self, name, attrs):
                 self.da.ind_ord = array_index_order_codes.code[
                     attrs["ArrayIndexingOrder"]]
             if "Dimensionality" in attrs:
-                self.da.num_dim = int(attrs["Dimensionality"])
-            for i in range(self.da.num_dim):
+                self.num_dim = int(attrs["Dimensionality"])
```

Comment:
Yeah, confusing--it's set on the parser. The parser then asserts that the explicit value matches what was found in the data, while parsing. It's never actually set on the image.